documents/entityrelationship/service.go

```diff
@@ -190,33 +190,93 @@ func (s service) DeriveEntityRelationshipResponse(model documents.Model) (*entit
 
 }
 
-// DeriveEntityData returns create response from entity model
-func (s service) DeriveEntityRelationshipData(doc documents.Model) (*entitypb.RelationshipData, error) {
-	er, ok := doc.(*EntityRelationship)
+// DeriveEntityRelationshipData returns create response from entity model
+func (s service) DeriveEntityRelationshipData(model documents.Model) (*entitypb.RelationshipData, error) {
+	er, ok := model.(*EntityRelationship)
 	if !ok {
 		return nil, documents.ErrDocumentInvalidType
 	}
 
-	return er.getClientData(), nil
+	return er.getRelationshipData(), nil
 }
 
-// DeriveFromUpdatePayload returns a new version of the old entity identified by identifier in payload
+// DeriveFromUpdatePayload returns a new version of the indicated Entity Relationship with a deleted access token
 func (s service) DeriveFromUpdatePayload(ctx context.Context, payload *entitypb.RelationshipPayload) (documents.Model, error) {
 	if payload == nil {
-		return nil, documents.ErrDocumentNil
+		return nil, documents.ErrPayloadNil
+	}
+
+	eID, err := hexutil.Decode(payload.Identifier)
+	if err != nil {
+		return nil, err
+	}
+
+	did, err := identity.StringsToDIDs(payload.TargetIdentity)
+	if err != nil {
+		return nil, err
+	}
+
+	selfDID, err := contextutil.AccountDID(ctx)
+	if err != nil {
+		return nil, errors.New("failed to get self ID")
+	}
+
+	id, err := s.repo.FindEntityRelationshipIdentifier(eID, selfDID, *did[0])
+	if err != nil {
+		return nil, err
+	}
+
+	r, err := s.GetCurrentVersion(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+
+	model, err := r.(*EntityRelationship).DeleteAccessToken(ctx, hexutil.Encode(did[0][:]))
```

Comment:
At the moment is fine, as there is no other operations over the ER. In the near future, this logic is not as agnostic as the `Derive` method should be.

Reply:
agree, in the future we can perhaps specify to DeriveFromRevokePayload which would differ from DeriveFromUpdatePayload